droid-command-line/src/main/java/uk/gov/nationalarchives/droid/command/action/ProfileRunCommand.java

```diff
@@ -100,18 +102,30 @@ public void onProgress(Integer progress) {
 
             profileManager.save(profile.getUuid(), Paths.get(destination), progressCallback);
             profileManager.closeProfile(profile.getUuid());
-        } catch (ProfileManagerException e) {
-            throw new CommandExecutionException(e);
-        } catch (InterruptedException e) {
+            terminateJsonArray();
+        } catch (ProfileManagerException | InterruptedException | IOException | SignatureFileException e) {
             throw new CommandExecutionException(e);
         } catch (ExecutionException e) {
             throw new CommandExecutionException(e.getCause());
-        } catch (IOException e) {
-            throw new CommandExecutionException(e);
-        } catch (SignatureFileException e) {
-            throw new CommandExecutionException(e);
         }
-        
+
+    }
+
+    public Writer getWriter(String outputFilePath) throws IOException {
+        if (outputFilePath != null && !outputFilePath.trim().isEmpty()) {
+            File outputFile = new File(outputFilePath);
+            return new FileWriter(outputFile, true);
+        }
+        return new PrintWriter(System.out);
+    }
+
+    public void terminateJsonArray() throws IOException {
```

Comment:
just a suggestion, to write a comment for this method's purpose, so others will know